Open Home page on closing last Tab in Tab Switcher

sauravchandra opened the pull request:
Fixes #1095 

Changes: 

1) Added Auto Transition to animate the layout change when showTabSwitcher() and hideTabSwitcher() is called.
2) Added a condition to check if all tabs are closed. If True, then open the Home page and hide Tab Switcher View, else update the Tab Switcher icon.

Screenshots/GIF for the change: 

![issue-fix](https://user-images.githubusercontent.com/10370619/54853084-cae1ef80-4d14-11e9-98d8-5385f994b4e8.gif)

abdulwd commented on app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java:
```diff
@@ -783,15 +790,18 @@ private void closeTab(int index) {
-      if (webViewList.size() == 0) {
         newTab(HOME_URL);
-      }
+        hideTabSwitcher();  //Fix for Issue #1095
```
Don't link issue numbers to code. This will unnecessarily blot the code base.

sauravchandra replied:
Sure. I will take care the next time.

abdulwd commented on app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java:
```diff
@@ -783,15 +790,18 @@ private void closeTab(int index) {
+    if (webViewList.size() == 0) {
+      openDefaultTab();
+    }
+    else {
```
`else` should be on the previous line. Follow the contributors' guidelines of this repo.

sauravchandra replied:
Okay.
```suggestion
    else {
```

sauravchandra replied:
Okay.

abdulwd commented on app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java:
```diff
@@ -234,6 +236,7 @@ public void onFinish() {
   private TextView tabSwitcherIcon;
   private TableDrawerAdapter tableDrawerAdapter;
   private RecyclerView tableDrawerRight;
+  private AutoTransition autoTransition;
```
Can we initialize this variable here itself instead of initializing it twice in the methods? Will it work?

sauravchandra replied:
Yes we can. I made the changes.

abdulwd commented on app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java:
```diff
@@ -783,15 +788,17 @@ private void closeTab(int index) {
-      }
+        hideTabSwitcher();
     }, 100);
   }
```
Can you add a blank line after this line so that there is a blank line after the method definition is complete?

sauravchandra replied:
Okay. I have made the change.

abdulwd commented on app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java:
```diff
@@ -783,15 +788,17 @@ private void closeTab(int index) {
   private void openDefaultTab() {
     new Handler().postDelayed(() -> {
-      if (webViewList.size() == 0) {
         newTab(HOME_URL);
```
Use 2 space indents instead of 4.

sauravchandra replied:
Sure. I made the changes. I will follow the coding guidelines on all future commits.

abdulwd commented:
Please squash all your commits into a single commit and force push the changes. Also, the animation is too fast can you slow it down a bit?

sauravchandra commented:
I have made changes to slow down the transition animation. I have squashed all commits into one and force pushed the changes. 

*Modified animation:*

![ezgif-2-b381edbd6d63](https://user-images.githubusercontent.com/10370619/54879869-5c13ac00-4e64-11e9-8f33-bb7ff52b5ee7.gif)

mhutti1 commented:
This has been fixed.
